File: 2021/12/08/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTilt(self, root: Optional[TreeNode]) -> int:
        self.tilt = 0
        def helper(root):
            if root == None:
                return 0

            l = helper(root.left)
            r = helper(root.right)

            if l > r:
                self.tilt += l - r
            else:
                self.tilt += r - l

            return l + r + root.val
        
        helper(root)
        
        return self.tilt

File: 2021/12/08/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findTilt(root *TreeNode) int {
	var tilt int
	helper(root, &tilt)
	return tilt
}

func helper(root *TreeNode, tilt *int) int {
	if root == nil {
		return 0
	}

	l := helper(root.Left, tilt)
	r := helper(root.Right, tilt)

	if l > r {
		*tilt += l - r
	} else {
		*tilt += r - l
	}

	return l + r + root.Val
}
